test/framework/modules.py

```diff
@@ -750,6 +750,110 @@ def test_exit_code_check(self):
             error_pattern = "Unable to locate a modulefile for 'nosuchmoduleavailableanywhere'"
         self.assertErrorRegex(EasyBuildError, error_pattern, self.modtool.load, ['nosuchmoduleavailableanywhere'])
 
+    def test_check_loaded_modules(self):
+        """Test check_loaded_modules method."""
+        # try and make sure we start with a clean slate
+        self.modtool.purge()
+
+        def check_loaded_modules():
+            "Helper function to run check_loaded_modules and check on stdout/stderr."
+            # there should be no errors/warnings by default is no (EasyBuild-generated) modules are loaded
+            self.mock_stdout(True)
+            self.mock_stderr(True)
+            self.modtool.check_loaded_modules()
+            stdout, stderr = self.get_stdout(), self.get_stderr()
+            self.mock_stdout(False)
+            self.mock_stderr(False)
+            self.assertEqual(stdout, '')
+            return stderr.strip()
+
+
+        # by default, having an EasyBuild module loaded is allowed
+        self.modtool.load(['EasyBuild/fake'])
+
+        # no output to stderr (no warnings/errors)
+        self.assertEqual(check_loaded_modules(), '')
+
+        self.modtool.unload(['EasyBuild/fake'])
+
+        # load OpenMPI module, which also loads GCC & hwloc
+        self.modtool.load(['OpenMPI/1.6.4-GCC-4.6.4'])
+
+        # default action is to print a clear warning message
+        stderr = check_loaded_modules()
+        patterns = [
```

Comment:
Part of the reason for copy-pasting the message in the tests is so that we know when the message gets changed (potentially by accident), so this is intentional.